File: scripts/download_images.py
import os
import time
import requests
from pathlib import Path

ACCESS_KEY = "YOUR_UNSPLASH_ACCESS_KEY"

image_config = {
    "home": {
        "hero": {
            "query": "chinese traditional temple architecture",
            "filename": "public/images/home/hero-bg.jpg"
        },
        "services_preview": {
            "query": "chinese meditation zen garden",
            "filename": "public/images/home/services-preview.jpg"
        },
        "team": [
            {
                "query": "asian senior master meditation",
                "filename": "public/images/team/master_1.jpg"
            },
            {
                "query": "chinese traditional medicine doctor",
                "filename": "public/images/team/master_2.jpg"
            },
            {
                "query": "feng shui master consultation",
                "filename": "public/images/team/master_3.jpg"
            }
        ]
    },
    "auth": {
        "login": {
            "query": "chinese ink painting landscape peaceful",
            "filename": "public/images/auth/login-bg.jpg"
        },
        "signup": {
            "query": "chinese calligraphy art meditation",
            "filename": "public/images/auth/signup-bg.jpg"
        }
    },
    "services": {
        "tools": [
            {
                "query": "chinese fortune telling tools",
                "filename": "public/images/services/tools/bazi-calc.jpg"
            },
            {
                "query": "chinese name calligraphy art",
                "filename": "public/images/services/tools/name-analysis.jpg"
            },
            {
                "query": "chinese zodiac fortune telling",
                "filename": "public/images/services/tools/daily-fortune.jpg"
            }
        ],
        "cases": [
            {
                "query": "feng shui home transformation",
                "filename": "public/images/services/cases/fengshui-case-1.jpg"
            },
            {
                "query": "chinese traditional office design",
                "filename": "public/images/services/cases/fengshui-case-2.jpg"
            }
        ]
    },
    "pricing": {
        "hero": {
            "query": "zen garden meditation stone",
            "filename": "public/images/pricing/hero.jpg"
        },
        "tiers": [
            {
                "query": "chinese meditation beginner peaceful",
                "filename": "public/images/pricing/starter.jpg"
            },
            {
                "query": "taichi practice in nature",
                "filename": "public/images/pricing/professional.jpg"
            },
            {
                "query": "chinese master teaching meditation",
                "filename": "public/images/pricing/master.jpg"
            }
        ]
    }
}

def download_image(query, filename):
    try:
        # Create directory if it doesn't exist
        os.makedirs(os.path.dirname(filename), exist_ok=True)
        
        # Construct the Unsplash API URL
        url = f"https://api.unsplash.com/photos/random?query={query}&orientation=landscape"
        headers = {"Authorization": f"Client-ID {ACCESS_KEY}"}
        
        # Get image URL from Unsplash
        response = requests.get(url, headers=headers)
        if response.status_code != 200:
            print(f"Error getting image URL for {query}: {response.status_code}")
            return
        
        image_url = response.json()["urls"]["regular"]
        
        # Download the image
        image_response = requests.get(image_url)
        if image_response.status_code != 200:
            print(f"Error downloading image for {query}: {image_response.status_code}")
            return
        
        # Save the image
        with open(filename, "wb") as f:
            f.write(image_response.content)
        
        print(f"Successfully downloaded: {filename}")
        
    except Exception as e:
        print(f"Error downloading {query}: {str(e)}")

def download_all_images():
    for category, items in image_config.items():
        for item_name, item_data in items.items():
            if isinstance(item_data, list):
                for item in item_data:
                    download_image(item["query"], item["filename"])
                    time.sleep(1)  # Rate limiting
            else:
                download_image(item_data["query"], item_data["filename"])
                time.sleep(1)  # Rate limiting

if __name__ == "__main__":
    download_all_images() 

File: scripts/download-images.js
const fs = require('fs');
const path = require('path');
const https = require('https');
const sharp = require('sharp');

// 使用Pixabay API
const PIXABAY_API_KEY = ''; // 需要填入有效的API key
const BASE_URL = 'https://pixabay.com/api/';

const IMAGE_CATEGORIES = {
  team: [
    { name: 'master_wang_1.jpg', query: 'chinese+master+meditation' },
    { name: 'master_li_1.jpg', query: 'meditation+teacher+asian' },
    { name: 'master_chen_1.jpg', query: 'feng+shui+master+asian' },
  ],
  services: {
    consultation: [
      { name: 'consultation_1.jpg', query: 'chinese+consultation+meeting' },
      { name: 'consultation_2.jpg', query: 'feng+shui+consultation' },
      { name: 'consultation_3.jpg', query: 'meditation+guidance+asian' },
    ],
    tools: [
      { name: 'calculator_1.jpg', query: 'chinese+calculator+traditional' },
      { name: 'compass_1.jpg', query: 'feng+shui+compass+chinese' },
      { name: 'calendar_1.jpg', query: 'chinese+calendar+traditional' },
    ],
    cases: [
      { name: 'case_study_1.jpg', query: 'feng+shui+home+interior' },
      { name: 'case_study_2.jpg', query: 'chinese+garden+traditional' },
      { name: 'case_study_3.jpg', query: 'meditation+space+zen' },
    ],
  },
};

async function searchImage(query) {
  const url = `${BASE_URL}?key=${PIXABAY_API_KEY}&q=${query}&image_type=photo&orientation=horizontal&per_page=3`;

  return new Promise((resolve, reject) => {
    https.get(url, (response) => {
      let data = '';

      response.on('data', (chunk) => {
        data += chunk;
      });

      response.on('end', () => {
        try {
          const result = JSON.parse(data);
          if (result.hits && result.hits.length > 0) {
            resolve(result.hits[0].largeImageURL);
          } else {
            reject(new Error('No images found'));
          }
        } catch (error) {
          reject(error);
        }
      });
    }).on('error', reject);
  });
}

async function downloadImage(url, filepath) {
  return new Promise((resolve, reject) => {
    https.get(url, (response) => {
      if (response.statusCode === 200) {
        const fileStream = fs.createWriteStream(filepath);
        response.pipe(fileStream);
        fileStream.on('finish', () => {
          fileStream.close();
          resolve(filepath);
        });
      } else {
        reject(new Error(`Failed to download image: ${response.statusCode}`));
      }
    }).on('error', reject);
  });
}

async function processImage(filepath) {
  try {
    await sharp(filepath)
      .resize(1200, 800, { fit: 'cover' })
      .jpeg({ quality: 85 })
      .toFile(filepath + '.processed');

    fs.renameSync(filepath + '.processed', filepath);
    console.log(`Processed: ${filepath}`);
  } catch (error) {
    console.error(`Error processing ${filepath}:`, error);
  }
}

async function ensureDirectoryExists(dirPath) {
  if (!fs.existsSync(dirPath)) {
    fs.mkdirSync(dirPath, { recursive: true });
  }
}

async function downloadAndProcessImages() {
  if (!PIXABAY_API_KEY) {
    console.error('Please provide a valid Pixabay API key');
    return;
  }

  // Process team images
  const teamDir = path.join('public', 'images', 'team');
  await ensureDirectoryExists(teamDir);

  for (const image of IMAGE_CATEGORIES.team) {
    const filepath = path.join(teamDir, image.name);
    try {
      const imageUrl = await searchImage(image.query);
      await downloadImage(imageUrl, filepath);
      await processImage(filepath);
      console.log(`Successfully processed: ${image.name}`);
    } catch (error) {
      console.error(`Error with ${image.name}:`, error);
    }
  }

  // Process service images
  for (const [category, images] of Object.entries(IMAGE_CATEGORIES.services)) {
    const categoryDir = path.join('public', 'images', 'services', category);
    await ensureDirectoryExists(categoryDir);

    for (const image of images) {
      const filepath = path.join(categoryDir, image.name);
      try {
        const imageUrl = await searchImage(image.query);
        await downloadImage(imageUrl, filepath);
        await processImage(filepath);
        console.log(`Successfully processed: ${image.name}`);
      } catch (error) {
        console.error(`Error with ${image.name}:`, error);
      }
    }
  }
}

downloadAndProcessImages().catch(console.error); 